Sources/Private/CoreAnimation/Animations/RectangleAnimation.swift

```diff
@@ -14,18 +14,48 @@ extension CAShapeLayer {
   {
     try addAnimation(
       for: .path,
-      keyframes: rectangle.size.keyframes,
-      value: { sizeKeyframe in
+      keyframes: try rectangle.combinedKeyframes(context: context).keyframes,
+      value: { keyframe in
         BezierPath.rectangle(
-          position: try rectangle.position
-            .exactlyOneKeyframe(context: context, description: "rectangle position").value.pointValue,
-          size: sizeKeyframe.sizeValue,
-          cornerRadius: try rectangle.cornerRadius
-            .exactlyOneKeyframe(context: context, description: "rectangle cornerRadius").value.cgFloatValue,
+          position: keyframe.position.pointValue,
+          size: keyframe.size.sizeValue,
+          cornerRadius: keyframe.cornerRadius.cgFloatValue,
           direction: rectangle.direction)
           .cgPath()
           .duplicated(times: pathMultiplier)
       },
       context: context)
   }
 }
+
+extension Rectangle {
+  /// Data that represents how to render a rectangle at a specific point in time
+  struct Keyframe {
+    let size: Vector3D
+    let position: Vector3D
+    let cornerRadius: Vector1D
+  }
+
+  /// Creates a single array of animatable keyframes from the separate arrays of keyframes in this Rectangle
+  func combinedKeyframes(context: LayerAnimationContext) throws-> KeyframeGroup<Rectangle.Keyframe> {
+    let combinedKeyframes = Keyframes.combinedIfPossible(size, position, cornerRadius) { untypedValues in
+      Keyframe(
+        size: untypedValues[0] as! Vector3D,
+        position: untypedValues[1] as! Vector3D,
+        cornerRadius: untypedValues[2] as! Vector1D)
```

Comment:
If we had variadic generics we could have the input parameters be `(KeyframeGroup<Vector3D>, KeyframeGroup<Vector3D>, KeyframeGroup<Vector1D>)` and have the output parameters be `(Vector3D, Vector3D, Vector1D). Since we can't represent that in the type system today, we have to just output an `[Any]` and cast the parameters manually.

This implementation is either correct and will always work, and is incorrect and would always crash. Since our test suite tests these methods very extensively, imo it's safe to assume that this will always succeed in production as opposed to doing an `as?` optional unwrap and handing a hypothetical failure case.